src/hotspot/share/code/codeCache.cpp

```diff
@@ -1559,6 +1559,33 @@ void CodeCache::log_state(outputStream* st) {
             unallocated_capacity());
 }
 
+void CodeCache::write_perf_map(outputStream* st) {
```

Comment:
Should this code (and changes in header files) be included for Linux only? (`#ifdef LINUX` like the change in diagnosticCommand.cpp)

Reply:
I'm not sure, I didn't want to add too much `#ifdef` mess. The code will compile on other platforms, it just won't be called. Better to add `#ifdef`s around all of it?

Reply:
Any reason not to have this dcmd supported on all platforms even though the output is really targeted for use with the perf tool on linux? Would a user ever have any other use for the output other than with the perf tool on linux?

Reply:
+#ifdef LINUX
+  DCmdFactory::register_DCmdFactory(new DCmdFactoryImpl<PerfMapDCmd>(full_export, true, false));
+#endif // LINUX

If this PR is for Linux only then I wonder if all changes have to be ifdef'ed the same or similar way.

Reply:
Sorry, I've overlooked that @YaSuenag posted similar comment.

Reply:
@plummercj I'm not sure: it's just a text file so could be easily consumed by tools other than perf. But I'm not aware of any tools on other platforms that could use it.